book-content/chapters/02-ide-superpowers.md

```diff
@@ -0,0 +1,579 @@
+TypeScript works the same no matter what IDE you use, but for this book we'll assume you're using Visual Studio Code (VS Code).
+
+When you open VS Code, the TypeScript server starts in the background. It will be active as long as you have a TypeScript file open.
+
+Let's take a look at some of the awesome VS Code features that are powered by the TypeScript server.
+
+## Autocomplete
+
+If I were to name the single TypeScript feature I couldn't live without, it would be autocomplete.
+
+TypeScript knows what type everything is inside your app. Because of that, it can give you suggestions when you're typing - speeding you up enormously.
+
+In the example below, just typing 'p' after `audioElement` brings up all the properties which start with 'p'.
+
+```typescript
+const audioElement = document.createElement("audio");
+
+audioElement.p; // play, pause, part etc
+```
+
+This is really powerful for exploring APIs you might not be familiar with, like the `HTMLAudioElement` API in this case.
+
+### Manually Triggering Autocomplete
+
+You'll often want to manually trigger autocomplete. In VS Code, the `Ctrl + Space` keyboard shortcut will show you a list of suggestions for what you're typing.
+
+For example, if you were adding an event listener to an element, you would see a list of available events:
+
+```typescript
+document.addEventListener(
+  "", // autocomplete here
+);
+```
+
+Hitting the `Ctrl + Space` shortcut with the cursor inside the quotes will show you a list of events that can be listened for:
+
+```
+DOMContentLoaded
+abort
+animationcancel
+...
+```
+
+If you wanted to narrow down the list to the events you were interested in, you could type "drag" before hitting `Ctrl + Space` and only the appropriate events would display:
```

Comment:
it sounds like "before hitting" is a requirement here but we get the same kind of filtering if we just start typing out things when the popup is already open